src/hooks/useProjectsLoader.tsx

```diff
@@ -0,0 +1,41 @@
+import { trap } from 'lib/trap'
+import { useState, useEffect } from 'react'
+import { ensureProjectDirectoryExists, listProjects } from 'lib/desktop'
+import { loadAndValidateSettings } from 'lib/settings/settingsUtils'
+import { Project } from 'lib/project'
+import { isDesktop } from 'lib/isDesktop'
+
+// Hook uses [number] to give users familiarity. It is meant to mimic a
+// dependency array, but is intended to only ever be used with 1 value.
```

Comment:
This comment explains why you made the API the way you did, but could you also explain what the hook is used for? My understanding is that it provides a self-refreshing version of the old `homeLoader` route loader? (I haven't gotten to that part yet but I bet it's been renamed, that's great)

Reply:
The hook is used so the Home route _doesn't_ have to reload. A full reload causes the whole view to flash due to all the content re-rendering.

This hook now exists to properly fetch a list of projects anywhere in the React application, separating itself from the router